runtime/runtime/src/lib.rs

```diff
@@ -1565,12 +1573,32 @@ impl Runtime {
         let apply_state = &mut processing_state.apply_state;
         let state_update = &mut processing_state.state_update;
         for signed_transaction in processing_state.transactions {
-            let (receipt, outcome_with_id) = self.process_transaction(
+            let tx_result = self.process_transaction(
                 state_update,
                 apply_state,
                 signed_transaction,
                 &mut processing_state.stats,
-            )?;
+            );
+            let (receipt, outcome_with_id) = match tx_result {
+                Ok(v) => v,
+                Err(e) => {
+                    if checked_feature!(
+                        "protocol_feature_relaxed_chunk_validation",
+                        RelaxedChunkValidation,
+                        processing_state.protocol_version
+                    ) {
+                        // TODO(new-tx): do we need to somehow report the invalid transactions
+                        //               to the callers? Currently the caller bails processing the
+                        //               entire chunk if we return an error here.
+                        // TODO(new-tx): if the transaction is invalid, do we still want to charge
+                        //               processing fees from somebody somewhere? Currently the
+                        //               state is simply rolled back.
+                        continue;
```

Comment:
As per discussion on Zulip, it seems that ignoring the transactions for now is fine, but we will want to revisit this question at a future date (possibly before stabilization.)

Reply:
So, I want to actually discuss this more. The errors generated by previous code here would `?`-up all the way up to chunk validation code. Which made me and @akhi3030 wonder how come it isn't a problem in *today*'s code that an invalid transaction *here* would cause problems due to transactions/receipts that have not been processed fully.

So first and foremost, there is an earlier place where the transactions are `verify_and_charge`d during chunk production: https://github.com/near/nearcore/blob/22259d0c2dd3dd2707795eb9d8b0cdb264d612f0/chain/chain/src/runtime/mod.rs#L861-L886.

So with that code linked up there my understanding is that invalid transactions can't actually make their way here. Unless the chunk producer applied different rules in `verify_and_charge`ing the transactions, making the chunk invalid. Which is what my integration test `test_invalid_transactions_dont_invalidate_chunk` represents. But now that I am writing this down, I am wondering if I accidentally actually arrived at an invalid understanding of the intended behaviour when writing down the test. For such invalid chunks should *still* be impossible, despite the removal of `new_transactions` checks…?

Reply:
I'll get back to thinking about this next week!